app/components/inputs/InputComponent.js

```diff
@@ -0,0 +1,188 @@
+      inputPhase: 0,
+      inputVal: "0.003000000",
+      inputMain: "0.00",
+      inputSub: "0000000"
+    };
+  }
+  componentDidMount() {
+    if (this.props.autoFocus) {
+      this.input.focus();
+    }
+  }
+  onInputFocus = (e) => {
+    const { onFocus } = this.props;
+    const { inputUnitDiv } = this.state;
+    var updatedInputUnitDiv = inputUnitDiv;
+    this.setState({ inputUnitDiv: updatedInputUnitDiv.classList.add("active") });
+    onFocus && onFocus(e);
+  };
+  onInputBlur = (e) => {
+    const { onBlur } = this.props;
+    const { inputUnitDiv } = this.state;
+    var updatedInputUnitDiv = inputUnitDiv;
+    this.setState({ inputUnitDiv: updatedInputUnitDiv.classList.remove("active") });
+    onBlur && onBlur(e);
+  };
+  onKeyDown = (e) => {
+    const { onKeyDownSubmit, onKeyDown } = this.props;
+    (e.keyCode === 13) && onKeyDownSubmit && onKeyDownSubmit(e);
+    onKeyDown && !e.defaultPrevented && onKeyDown(e);
+  }
+  onKeyDown2 = (e) => {
+    if (e.keyCode === 13) {
+      e.preventDefault();
+    }
+    const { onKeyDownSubmit, onKeyDown } = this.props;
+    (e.keyCode === 13) && onKeyDownSubmit && onKeyDownSubmit(e);
+    onKeyDown && !e.defaultPrevented && onKeyDown(e);
+  }
+  onChange = () => {
+    this.setState({ inputPhase: 1 });
+    var number = this.state.inputMain + this.state.inputSub;
+    this.setState({ inputVal: number });
+  };
+  onChange2 = (e) => {
+    this.setState({ inputPhase: 0 });
+    var number = e.target.textContent;
+    var number_f = parseFloat(number);
+    var left_part = number_f.toFixed(2);
+    var right_part_raw = number_f - left_part;
+    var right_part_rounded = parseFloat(String(right_part_raw.toFixed(10)).slice(0, -1));
+
+    var right_part = right_part_rounded * 100000000;
+
+    // Used to compare value later
+    var prepad = right_part_rounded * 100000000;
+
+    right_part = right_part.toFixed(0);
+
+    // Deduce rounding errors
+    if (Math.floor(prepad) < right_part && Math.floor(prepad) == (parseInt(right_part)) - 1 ) {
+      right_part = (parseInt(right_part)) - 1;
+    }
+
+    // Deduce rounding errors (upto 10 decimal places)
+    else if (Math.floor(right_part_raw * 100000000) < right_part && Math.floor(right_part_raw * 100000000) == (parseInt(right_part)) - 1) {
+      right_part = (parseInt(right_part)) - 1;
+    }
+
+    // Pad for numbers, so 0032 decimal will not turn to 32
+    right_part = right_part.toString().padStart(6, "0");
+
+    if (right_part <= 0) {
+      right_part = "000000";
+    }
+
+    // Reset back to numbers
+    if (isNaN(left_part) == true || isNaN(right_part) == true) {
+      left_part = "0.00";
+      right_part = "000000";
+    }
+
+    this.setState({ inputMain: left_part });
+    this.setState({ inputSub: right_part });
+
+    // Send to other older previous unstyled component and trigger event
+    document.getElementById("parsedInput").value = number_f;
```

Comment:
Still using getElementById...

Reply:
That one section is to mirror the changes for validation and submit event. The thing I made is a different element that links to this original input and this part sends the new element changes to the old element. I am unsure how to change this because of the 'value' variable of the original input (<input value={value}) not sure where exactly that 'value' variable comes from and how it is changed and how the validation works with this 'value'.

Reply:
The `value` attribute being passed to the original `<input>` element comes from whoever is using this input component: you can see that it's coming from the props being passed to this component (I'm going to mark it with a comment so it can be obvious).

(apologies if you already know what I'm about to show; it's just that it seems you're not so experienced in React patterns)

It's common in React applications to have an input lifecycle that roughly looks like this:

```javascript
class ComponentInput extends React.Component {
  render() {
    return <input value={this.props.value} onChange={e => this.props.onChange(e.target.value)} />
  }
}

class ComponentThatUsesInput extends React.Component {
  onChage(newValue) {
    this.setState({ value: newValue });
  }

  render() {
    return <ComponentInput value={this.state.value} onchanve={this.onChange} />
  }
}

```
Notice that the content of `ComponentInput` (the contents of the `value` prop) is actually controlled by the _user_ of `ComponentInput` (instances of `ComponentThatUsesInput` class). 

Another way of looking at this is that `value` is a _state_ of `ComponentThatUsesInput` while it's a _prop_ of `ComponentInput`.

Yet another way of looking is that the _owner_ of whatever is in `value` is `ComponentThatUsesInput`, while `ComponentInput` merely displays its content and signals intentions to modify it (the `onChange` event) up the stack.

If you look at Decrediton's original `Input` component (and any place where it's used, such as the [message text input](https://github.com/decred/decrediton/blob/master/app/components/views/SecurityPage/VerifyMessage/Form.js#L77)) you'll see that it's following this pattern.

When you have a situation where you need to do some intermediate processing in a component, what you need to do is modify this "content propagation loop" such that the lower component processes the content then upstreams the modified value. For example, let's say that you want to always prefix the value with spaces, you'd modify the above code in the following way:

```javascript
class ComponentInput extends React.Component {
  onChange(newValue) {
    this.props.onChange("    "+newValue.trim())
  }
  render() {
    return <input value={this.props.value} onChange={e => this.onChange(e.target.value)} />
  }
}
```

Notice the following points:
- We *only* needed to change `ComponentInput`
- In `render()`, instead of directing the `onChange` event to whatever is passed via props, we hooked a local function to it
- In that function we preprocess the new value before signaling the change up the component stack

So that's roughly what you need to do in order to create a new component that responds in the appropriate way: the component receives a value via its props, reacts locally to events generating a new value which is propagated up via an event prop.

More concretely for the task at hand: in the `render()` method of your new component, you process `this.props.value` in whatever way you want (split it or whatever) and display its various parts. You then hook up individual `onChange` (or `onKeyDown` or whatever events are needed) to generate a new "candidate" value  that is passed up via `props.onChange`.